apps/wallet-mobile/src/features/ReviewTx/common/hooks/useLegacyOnConfirm.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react'
+
+import {ConfirmTxWithHwModal} from '../../../../components/ConfirmTxWithHwModal/ConfirmTxWithHwModal'
+import {ConfirmTxWithOsModal} from '../../../../components/ConfirmTxWithOsModal/ConfirmTxWithOsModal'
+import {ConfirmTxWithSpendingPasswordModal} from '../../../../components/ConfirmTxWithSpendingPasswordModal/ConfirmTxWithSpendingPasswordModal'
+import {useModal} from '../../../../components/Modal/ModalContext'
+import {YoroiSignedTx, YoroiUnsignedTx} from '../../../../yoroi-wallets/types/yoroi'
+import {useSelectedWallet} from '../../../WalletManager/common/hooks/useSelectedWallet'
+import {useNavigateTo} from './useNavigateTo'
+import {OnConfirm} from './useOnConfirm'
+import {useStrings} from './useStrings'
+
+export const useLegacyOnConfirm = ({
+  unsignedTx,
+  onSuccess,
+  onError,
+  onNotSupportedCIP1694,
+  onCIP36SupportChange,
+}: OnConfirm & {
+  unsignedTx?: YoroiUnsignedTx | null
+  onNotSupportedCIP1694?: (() => void) | null
+  onCIP36SupportChange?: ((isCIP36Supported: boolean) => void) | null
+}) => {
+  const {meta} = useSelectedWallet()
+  const {openModal, closeModal} = useModal()
+  const strings = useStrings()
+  const navigateTo = useNavigateTo()
+
+  const handleOnSuccess = (signedTx: YoroiSignedTx) => {
+    if (onSuccess) {
+      onSuccess({signedTx})
+      return
+    }
+
+    navigateTo.showSubmittedTxScreen()
+  }
+  const handleOnError = (error: unknown) => {
+    if (onError) {
+      onError(error)
+      return
+    }
+    navigateTo.showFailedTxScreen()
+  }
+
+  const legacyOnConfirm = () => {
+    if (meta.isHW && unsignedTx) {
+      openModal(
+        strings.signTransaction,
+        <ConfirmTxWithHwModal
+          onCancel={closeModal}
+          unsignedTx={unsignedTx}
+          onSuccess={handleOnSuccess}
+          onNotSupportedCIP1694={() => {
+            if (onNotSupportedCIP1694) {
+              closeModal()
+              onNotSupportedCIP1694()
+            }
+          }}
+          onCIP36SupportChange={onCIP36SupportChange ?? undefined}
+        />,
+        400,
+      )
+      return
+    }
+
+    if (!meta.isHW && !meta.isEasyConfirmationEnabled && unsignedTx) {
+      openModal(
+        strings.signTransaction,
+        <ConfirmTxWithSpendingPasswordModal
+          unsignedTx={unsignedTx}
+          onSuccess={handleOnSuccess}
+          onError={handleOnError}
+        />,
+      )
+      return
+    }
+
+    if (!meta.isHW && meta.isEasyConfirmationEnabled && unsignedTx) {
+      openModal(
+        strings.signTransaction,
+        <ConfirmTxWithOsModal unsignedTx={unsignedTx} onSuccess={handleOnSuccess} onError={handleOnError} />,
+      )
+      return
+    }
+
+    throw new Error('useLegacyOnConfirm:: invalid state')
+  }
```

Comment:
I think it can be simplified

```suggestion
const legacyOnConfirm = () => {
  if (!unsignedTx) {
    throw new Error('useLegacyOnConfirm:: unsignedTx is required')
  }

  const { isHW, isEasyConfirmationEnabled } = meta

  if (isHW) {
    openModal(HW)...
    return
  }

  if (!isEasyConfirmationEnabled) {
    openModal(WithPassword)
    return
  }
  
  openModal(WithOS)
}
```